shared/src/types/ethereum_events/vote_extensions.rs

```diff
@@ -0,0 +1,299 @@
+//! Contains types necessary for processing Ethereum events
+//! in vote extensions.
+
+use std::collections::HashSet;
+
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+use eyre::{eyre, Result};
+use num_rational::Ratio;
+
+use super::EthereumEvent;
+use crate::proto::Signed;
+use crate::types::address::Address;
+use crate::types::key::common::Signature;
+use crate::types::storage::BlockHeight;
+
+/// This struct will be created and signed over by each
+/// validator as their vote extension.
+#[derive(Debug, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
+pub struct VoteExtension {
```

Comment:
This is fine for now, but I will move this struct to a different place later since vote extesnions have uses other than just the ethereum bridge.

Reply:
But I will have to think about how to decompress digests in that case, so I'm not going to spend brain power on it now.